src/prompt/types.ts

```diff
@@ -1,17 +1,21 @@
 import type { z } from 'zod';
 
 export namespace Prompt {
+  export type ChainParams = Record<string, any> | void;
+  export type ChainResult = string | Record<string, any>;
+
   /**
    * A prompt chain that coordinates the template, functions, and validator.
    */
-  export type Chain<Args extends Record<string, any>, Result extends any> = (
-    args: Args
-  ) => Promise<Result>;
+  export type Chain<
+    Params extends ChainParams = void,
```

Comment:
Typing-wise, I think it's important to be able to invoke a chain without any params, and then have the result default to a `string`. This matches what I was doing for `agentic` 1:1, with the only real difference being that agentic supported any type of json output, not just string | object.